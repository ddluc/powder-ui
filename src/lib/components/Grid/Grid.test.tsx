import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Grid, Cell, Area, GridProps, CellProps, AreaProps } from './index';

afterEach(cleanup);

// Define unit tests
describe('Grid', () => {
  const args: GridProps = {
    columns: '80px 1fr 250px 2fr',
    rows: 'repeat(3, 60px)',
    rowGap: '20px',
    columnGap: '20px',
    justifyContent: 'flex-start',
    alignContent: 'flex-start',
    alignItems: 'center'
  };

  it('should render with defaults', () => {
    const result = render(<Grid />);
    const component = result.container.firstChild;
    expect(component).toMatchSnapshot();
  });

  it('should render with defined props', () => {
    const result = render(<Grid {...args} />);
    const component = result.container.firstChild;
    expect(component).toMatchSnapshot();
  });

  it('should render with template areas', () => {
    const areas = [
      'header header header header',
      'leftpanel content content rightpanel',
      '. footer footer .'
    ];
    const result = render(<Grid {...args} areas={areas} />);
    const component = result.container.firstChild;
    expect(component).toMatchSnapshot();
  });
});

describe('Cell', () => {
  const args: CellProps = {
    column: {
      start: 1,
      end: 'span 2'
    },
    row: {
      start: 2,
      end: 5
    },
    justify: 'end',
    align: 'stretch'
  };
  it('should render with defined props', () => {
    const result = render(<Cell {...args} />);
    const component = result.container.firstChild;
    expect(component).toMatchSnapshot();
  });
});

describe('Area', () => {
  const args: AreaProps = {
    area: 'leftpanel',
    justify: 'center',
    align: 'end'
  };
  it('should render with defined props', () => {
    const result = render(<Area {...args} />);
    const component = result.container.firstChild;
    expect(component).toMatchSnapshot();
  });
});
