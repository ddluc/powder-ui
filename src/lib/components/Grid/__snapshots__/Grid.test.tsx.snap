// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Area should render with defined props 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.c1 {
  grid-area: leftpanel;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
}

<div
  class="c0 c1"
/>
`;

exports[`Cell should render with defined props 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.c1 {
  grid-column-start: 1;
  grid-column-end: span 2;
  grid-row-start: 2;
  grid-row-end: 5;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

<div
  class="c0 c1"
/>
`;

exports[`Grid should render with defaults 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.c1 {
  display: grid;
}

<div
  class="c0 c1"
/>
`;

exports[`Grid should render with defined props 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.c1 {
  display: grid;
  grid-template-columns: 80px 1fr 250px 2fr;
  grid-template-rows: repeat(3,60px);
  -webkit-column-gap: 20px;
  column-gap: 20px;
  row-gap: 20px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0 c1"
/>
`;

exports[`Grid should render with template areas 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.c1 {
  display: grid;
  grid-template-columns: 80px 1fr 250px 2fr;
  grid-template-rows: repeat(3,60px);
  grid-template-areas: "header header header header" "leftpanel content content rightpanel" ". footer footer .";
  -webkit-column-gap: 20px;
  column-gap: 20px;
  row-gap: 20px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0 c1"
/>
`;
